applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h

```diff
@@ -0,0 +1,396 @@
+#define DAMPING_UTILITIES_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+#include <boost/python.hpp>
+#include <boost/numeric/ublas/matrix.hpp>
+#include <boost/numeric/ublas/vector.hpp>
+#include <boost/numeric/ublas/io.hpp>
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "../../kratos/includes/define.h"
+#include "../../kratos/processes/process.h"
+#include "../../kratos/includes/node.h"
+#include "../../kratos/includes/element.h"
+#include "../../kratos/includes/model_part.h"
+#include "../../kratos/includes/kratos_flags.h"
+#include "../../kratos/spatial_containers/spatial_containers.h"
+#include "../../kratos/utilities/timer.h"
+#include "../../kratos/processes/node_erase_process.h"
+#include "../../kratos/utilities/binbased_fast_point_locator.h"
+#include "../../kratos/utilities/normal_calculation_utils.h"
+#include "shape_optimization_application.h"
+#include "damping_function.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+    typedef boost::python::extract<ModelPart&> extractModelPart;
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+
+*/
+
+class DampingUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    // Type definitions for better reading later
+    typedef array_1d<double,3> array_3d;
+    typedef Node < 3 > NodeType;
+    typedef Node < 3 > ::Pointer NodeTypePointer;
+    typedef std::vector<NodeType::Pointer> NodeVector;
+    typedef std::vector<NodeType::Pointer>::iterator NodeIterator;
+    typedef std::vector<double> DoubleVector;
+    typedef std::vector<double>::iterator DoubleVectorIterator;
+    typedef ModelPart::ConditionsContainerType ConditionsArrayType;
+
+    // Type definitions for tree-search
+    typedef Bucket< 3, NodeType, NodeVector, NodeTypePointer, NodeIterator, DoubleVectorIterator > BucketType;
+    typedef Tree< KDTreePartition<BucketType> > KDTree;    
+
+    /// Pointer definition of DampingUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(DampingUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    DampingUtilities( ModelPart& modelPartToDamp, boost::python::dict subModelPartsForDamping, Parameters& optimizationSettings )
```

Comment:
Quoting the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide), section _Arguments_, "Write arguments with a UpperCamelCase format."
E.g. `rModelPartToDamp` instead of `modelPartToDamp`